### PR TITLE
ENH: Add parameter for depth

### PR DESCRIPTION
**Pull request by Eric89GXL**
Addresses https://github.com/vispy/vispy/issues/953

Ready for review/merge from my end

### CODE REVIEW

File: vispy/scene/cameras/base_camera.py [hunk 1]
```diff
@@ -79,28 +79,35 @@ def __init__(self, interactive=True, flip=None, up='+z', parent=None,
...
         # Set parameters. These are all not part of the "camera state"
         self.interactive = bool(interactive)
         self.flip = flip if (flip is not None) else (False, False, False)
         self.up = up
 
-    def _get_depth_value(self):
-        """ Get the depth value to use in orthographic and perspective projection
+    @property
+    def depth_value(self):
+        """The depth value to use  in orthographic and perspective projection
```
Comment (campagnola) [inline vispy/scene/cameras/base_camera.py]: What IS the depth value? What are its practical effects? I assume it somehow relates to the distance to the far clipping plane?

Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: See your comments here: https://github.com/vispy/vispy/issues/953 :)

Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: My understanding we need to have a way to allow people to have a large range, or high precision around zero, and the `depth_value` relates to that.

Comment (campagnola) [inline vispy/scene/cameras/base_camera.py]: Ok, but we need a concise definition. 
I guess I'll have to READ the CODE! UGH!

Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: This is where it actually gets used in `PerspectiveCamera`:
```
        d = self.depth_value
        if self._fov == 0:
            self._projection.set_ortho(-0.5*fx, 0.5*fx, -0.5*fy, 0.5*fy, -d, d)
            self._actual_distance = self._distance or 0.0
        else:
            # Figure distance to center in order to have correct FoV and fy.
            # Use that auto-distance, or the given distance (if not None).
            fov = max(0.01, self._fov)
            dist = fy / (2 * math.tan(math.radians(fov)/2))
            self._actual_distance = dist = self._distance or dist
            val = math.sqrt(d*10)
            self._projection.set_perspective(fov, fx/fy, dist/val, dist*val)
```

Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: So I guess it sets the limits for the orthgraphic or perspective transforms...?

Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: And that above is from `Base3DCamera`, not `PerspectiveCamera` (which uses a slightly different definition)

Comment (campagnola) [inline vispy/scene/cameras/base_camera.py]: For orthographic, `depth_value` is the *distance* between the near and far clipping planes. For perspective, it is the *ratio* between the near and far clipping plane distances.

Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: You want to add that comment, or shall I?

Comment (campagnola) [inline vispy/scene/cameras/base_camera.py]: Go ahead, then I'll merge.

File: vispy/scene/cameras/base_camera.py [hunk 2]
```diff
@@ -79,28 +79,39 @@ def __init__(self, interactive=True, flip=None, up='+z', parent=None,
...
+
+        For orthographic projections, ``depth_value`` is the distance between
+        the near and far clipping planes. For perspective projections, it is
+        the ratio between the near and far clipping plane distances.
```
Comment (Eric89GXL) [inline vispy/scene/cameras/base_camera.py]: Added these lines

**Comment by campagnola** [comment]
Grazie!

**Pull request merged**